tests/Tests/500__jupyterhub/jupyterhub-user-access.robot

```diff
@@ -2,59 +2,170 @@
+    Capture Page Screenshot    perm_admin.png
+    Logout Via Button
+    Go To RHODS Dashboard
+    Launch JupyterHub From RHODS Dashboard Link
+    Run Keyword And Continue On Failure   Login Verify Access Level    ${TEST_USER_3.USERNAME}
+    ...                                   ${TEST_USER_3.PASSWORD}    ${TEST_USER_3.AUTH_TYPE}    user
+    Capture Page Screenshot    perm_user.png
+    Logout Via Button
+
+Check Standard Access Configuration Works As Expected
+    [Documentation]    Checks if the standard access configuration
+    ...                works as expected in JH
+    Go To RHODS Dashboard
+    Launch JupyterHub From RHODS Dashboard Link
+    Handle Bad Gateway Page
+    Run Keyword And Continue On Failure   Login Verify Access Level  ${TEST_USER_2.USERNAME}
+    ...                                   ${TEST_USER_2.PASSWORD}    ${TEST_USER_2.AUTH_TYPE}    admin
+    Capture Page Screenshot    perm_admin_std.png
+    Logout Via Button
+    Go To RHODS Dashboard
+    Launch JupyterHub From RHODS Dashboard Link
+    Run Keyword And Continue On Failure   Login Verify Access Level    ${TEST_USER_4.USERNAME}
+    ...                                   ${TEST_USER_4.PASSWORD}    ${TEST_USER_4.AUTH_TYPE}    user
+    Capture Page Screenshot    perm_user_std.png
+    Logout Via Button
+
+Apply New Groups Config Map
+    [Documentation]    Changes the rhods-groups config map to set the new access configuration
+    OpenShiftCLI.Patch    kind=ConfigMap
+    ...                   src={"data":{"admin_groups": "${CUSTOM_ADMINS_GROUP}","allowed_groups": "${CUSTOM_USERS_GROUP}"}}
+    ...                   name=rhods-groups-config   namespace=redhat-ods-applications  type=merge
+    OpenShiftCLI.Patch    kind=ConfigMap
+    ...                   src={"metadata":{"labels": {"opendatahub.io/modified": "true"}}}
+    ...                   name=rhods-groups-config   namespace=redhat-ods-applications  type=merge
+
+Restore RHODS Standard Groups Config Map
+    [Documentation]    Restores the standard rhods-groups config map
+    OpenShiftCLI.Patch    kind=ConfigMap
+    ...                   src={"data":{"admin_groups": "${STANDARD_ADMINS_GROUP}","allowed_groups": "${STANDARD_USERS_GROUP}"}}
+    ...                   name=rhods-groups-config   namespace=redhat-ods-applications  type=merge
+    OpenShiftCLI.Patch    kind=ConfigMap
+    ...                   src={"metadata":{"labels": {"opendatahub.io/modified": "${STANDARD_GROUPS_MODIFIED}"}}}
+    ...                   name=rhods-groups-config   namespace=redhat-ods-applications  type=merge
+
+Restore Standard RHODS Groups Configuration
+    [Documentation]    Restores the standard RHODS access configuration
+    Restore RHODS Standard Groups Config Map
+    Rollout JupyterHub
+    Go To    ${OCP_CONSOLE_URL}
+    Add Test Users Back To RHODS Standard Groups
+    Remove Test Users From Custom Groups
+    Delete Custom Groups
+    Check Standard Access Configuration Works As Expected
```

Comment:
Should this check be in the Test Teardown? Maybe it makes more sense to have a test for this by itself:

- ODS-503: Verify Access To JupyterHub Using Custom Access Groups
- ODS-XXX:  Verify Access To JupyterHub Using Default Access Groups

Reply:
yeah in fact `Restore Standard RHODS Groups Configuration` is the teardown

Even if I created a separated test case, I would have run this check in ODS-503 to ensure the test case leave the environment as it was at the begininning

Reply:
it wants to be just a extra safety check since the test case changes some main setting of the RHODS installation..it doesn't add much complexity. i would like to keep it